troposphere/static/js/components/images/list/common/ImageListCard.react.js

```diff
@@ -14,26 +16,44 @@ import stores from "stores";
 export default React.createClass({
     displayName: "ImageListCard",
 
+    mixins: [Router.State],
+
     propTypes: {
         image: React.PropTypes.instanceOf(Backbone.Model).isRequired
     },
 
-    render: function() {
+    onCardClick() {
+        RouterInstance.getInstance()
+            .transitionTo("image-details",{
+                imageId: this.props.image.id
+            });
+    },
+
+    render() {
         var image = this.props.image,
             hasLoggedInUser = context.hasLoggedInUser(),
             type = stores.ProfileStore.get().get("icon_set"),
             imageTags = stores.TagStore.getImageTags(image),
             imageCreationDate = moment(image.get("start_date"))
                 .tz(globals.TZ_REGION)
-                .format("MMM Do YYYY hh:mm a z"),
+                .format("MMM Do YY hh:mm "),
             converter = new Showdown.Converter(),
             description = image.get("description");
+            name = image.get('name');
 
+        if (imageTags.length > 10) {
+            imageTags = imageTags.first(10);
+        }
+        if (name.length > 30) {
+            name = name.substring(0,30) + " ..."
+        }
```

Comment:
You'll notice in this method there is a cadence of:
```
if (...) {
     <set some var>
}
if (...) {
     <set some var>
}
```
When I read it I have to visually scan and think about, "What if the condition is false and the var is never set?". A much better pattern is as follows:
```
let x = <some default value>
if (...) {
    x = <some other value>
}
let y;
if (...) {
    y = <some value>
} else {
    y = <some other value>
}
```